### PR TITLE
WIP: Add ppc64le support to Quay Operator

### PR DESCRIPTION
**Pull request by jcho02**
Signed-off-by: Jason Cho <jason.cho2@ibm.com>

ppc64le support

### CODE REVIEW

File: .github/workflows/build-and-publish.yaml [hunk 1]
```diff
@@ -70,7 +70,11 @@ jobs:
       - name: Install yq
         env:
           VERSION: v4.14.2
-          BINARY: yq_linux_amd64
+          override-build: |
```
Comment (dmage) [inline .github/workflows/build-and-publish.yaml]: `override-build` is not mentioned by [GitHub Actions Docs](https://docs.github.com/en/search?query=%22override-build%22), what is it?

Comment (jcho02) [reply .github/workflows/build-and-publish.yaml]: Updated the workflow files

File: .github/workflows/build-and-publish.yaml [hunk 2]
```diff
@@ -67,14 +67,24 @@ jobs:
...
+        if: ${{ "$(uname -m)" == "amd64" }}
         env:
           VERSION: v4.14.2
           BINARY: yq_linux_amd64
         run: |
           wget https://github.com/mikefarah/yq/releases/download/${VERSION}/${BINARY} -O /usr/local/bin/yq
           chmod +x /usr/local/bin/yq
 
+      - name: Install yq for ppc64le
+        if: ${{ "$(uname -m)" == "ppc64le" }}
```
Comment (dmage) [inline .github/workflows/build-and-publish.yaml]: We use GitHub-hosted runners, and they are available [only for x86_64](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).

Also you cannot use shell commands in GitHub expressions `${{ ... }}`, see [GitHub Docs for Expressions](https://docs.github.com/en/actions/learn-github-actions/expressions).

File: Dockerfile [hunk 3]
```diff
@@ -10,7 +10,11 @@ COPY apis/ apis/
 COPY controllers/ controllers/
 COPY pkg/ pkg/
 
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -mod vendor -o manager main.go
+RUN if [ ${ARCH}="ppc64le" ] ; then \
```
Comment (dmage) [inline Dockerfile]: You can simply remove `GOOS=linux GOARCH=amd64`.

File: Makefile [hunk 4]
```diff
@@ -19,7 +26,7 @@ test: generate fmt vet manifests
 
 test-e2e:
 	mkdir -p ./bin
-	curl -L https://github.com/kudobuilder/kuttl/releases/download/v0.11.1/kubectl-kuttl_0.11.1_linux_x86_64 -o ./bin/kuttl;
+	curl -L https://github.com/kudobuilder/kuttl/releases/download/v0.11.1/kubectl-kuttl_0.11.1_linux_${ARCH} -o ./bin/kuttl;
```
Comment (dmage) [inline Makefile]: Don't complicate things, use `_linux_$(uname -m)`

**Comment by dmage** [comment]
Did you try to build the docker image on ppc64le?

File: .github/workflows/build-and-publish.yaml (hunk 2 above)
Comment (jcho02) [reply .github/workflows/build-and-publish.yaml]: How would you like us to proceed regarding workflows if GH runners are only available for x86_64.

File: Dockerfile (hunk 3 above)
Comment (modassarrana89) [reply Dockerfile]: @dmage why can't we use --platform concept . 
```
FROM --platform=${BUILDPLATFORM:-linux/amd64} golang:1.17 as builder
ARG BUILDPLATFORM 
ARG TARGETARCH
RUN CGO_ENABLED=0 GOOS=linux GOARCH=${TARGETARCH} GO111MODULE=on go build -mod vendor -o manager main.go
```

I am actually planning to support quay-operator on Z & doing the same changes 

For makefile
------------
```
export DOCKER_CLI_EXPERIMENTAL=enabled

.PHONY: build # Build the container image
docker-build:
        @docker buildx create --use --name=crossplat --node=crossplat && \
        docker buildx build --platform linux/s390x,linux/amd64,linux/ppc64le  --tag $(IMG) .
```

File: Makefile (hunk 4 above)
Comment (modassarrana89) [reply Makefile:29]:
```
arch=$(shell uname -m)
...

curl -L https://github.com/kudobuilder/kuttl/releases/download/v0.11.1/kubectl-kuttl_0.11.1_linux_$(arch) -o ./bin/kuttl;
...
```

**Comment by jcho02** [comment]
Docker image builds on ppc64le and deploys on ppc64le supported OCP clusters.
WIP tag for QuayRegistry's

File: Makefile [hunk 5]
```diff
@@ -63,7 +62,7 @@ generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
 # Build the docker image
-docker-build: test
+docker-build:
```
Comment (yussufsh) [inline Makefile]: Why are we removing test target?

Comment (jcho02) [reply Makefile]: Oops, sorry removed that while testing locally. Test target is back

**Pull request closed without merging**